heron/spi/src/java/com/twitter/heron/spi/utils/SchedulerUtils.java

```diff
@@ -431,16 +434,45 @@ public static boolean curlAndExtractPackage(
   /**
    * Get a resource that requires the maximum amount of ram, cpu and disk
    */
-  public static Resource getMaxRequiredResource(PackingPlan packingPlan) {
+  public static PackingPlan getHomogenizedContainerPlan(PackingPlan packingPlan) {
     double maxCpu = 0;
     long maxRam = 0;
     long maxDisk = 0;
     for (PackingPlan.ContainerPlan containerPlan : packingPlan.getContainers()) {
-      maxCpu = Math.max(maxCpu, containerPlan.getResource().getCpu());
-      maxRam = Math.max(maxRam, containerPlan.getResource().getRam());
-      maxDisk = Math.max(maxDisk, containerPlan.getResource().getDisk());
+      maxCpu = Math.max(maxCpu, containerPlan.getRequiredResource().getCpu());
+      maxRam = Math.max(maxRam, containerPlan.getRequiredResource().getRam());
+      maxDisk = Math.max(maxDisk, containerPlan.getRequiredResource().getDisk());
     }
 
-    return new Resource(maxCpu, maxRam, maxDisk);
+    Resource maxResource = new Resource(maxCpu, maxRam, maxDisk);
+    Set<PackingPlan.ContainerPlan> updatedContainers = new HashSet<>();
+    for (PackingPlan.ContainerPlan container : packingPlan.getContainers()) {
+      PackingPlan.ContainerPlan updatedContainer = new PackingPlan.ContainerPlan(container.getId(),
```

Comment:
This seems like a useful method to add directly to `ContainerPlan`:
```
public PackingPlan.ContainerPlan cloneWithScheduledResource(Resource resource);
```
We did something similar with `Resource.cloneWithRam`. Then the new constructor on ContainerPlan could be private and we use the clone method instead.